core/net/mac/tsch/tsch-adaptive-timesync.c

```diff
@@ -86,12 +86,16 @@ timesync_entry_add(int32_t val, uint32_t time_delta)
 static void
 timesync_learn_drift_ticks(uint32_t time_delta_asn, int32_t drift_ticks)
 {
-  /* should fit in 32-bit unsigned integer */
-  uint32_t time_delta_ticks = time_delta_asn * tsch_timing[tsch_ts_timeslot_length];
+  /* should fit in a 32-bit integer */
+  int32_t time_delta_ticks = time_delta_asn * tsch_timing[tsch_ts_timeslot_length];
   int32_t real_drift_ticks = drift_ticks + compensated_ticks;
   int32_t last_drift_ppm = (int32_t)((int64_t)real_drift_ticks * TSCH_DRIFT_UNIT / time_delta_ticks);
 
   drift_ppm = timesync_entry_add(last_drift_ppm, time_delta_ticks);
+
+  TSCH_LOG_ADD(tsch_log_message,
+      snprintf(log->message, sizeof(log->message),
+          "drift %ld", drift_ppm / 256));
```

Comment:
There are some big differences here:
- `dr` prints drift correction in ticks, this prints the actual drift in ppm
- the drift value is only printed for packets that received from the time source node and used for time correction (and even then not always - not if the delta time is too small). There is no function to estimate the drift ppm of arbitrary neighbors in the network.
So for most of packets the ppm drift will just be "undetermined".
Also, the buffer for log message lines is quite but (48 bytes) and would need to be increased further if we want print more stuff for each tx/rx message.
I just find the printing this drift quite useful, especially for people starting with TSCH.
Maybe can somehow disable this by default, if you think it's too much?